hummingbot/market/eterbase/eterbase_order_book_tracker.py

```diff
@@ -0,0 +1,229 @@
+        super().__init__(data_source_type=data_source_type)
+
+        self._ev_loop: asyncio.BaseEventLoop = asyncio.get_event_loop()
+        self._data_source: Optional[OrderBookTrackerDataSource] = None
+        self._order_book_snapshot_stream: asyncio.Queue = asyncio.Queue()
+        self._order_book_diff_stream: asyncio.Queue = asyncio.Queue()
+        self._process_msg_deque_task: Optional[asyncio.Task] = None
+        self._past_diffs_windows: Dict[str, Deque] = {}
+        self._order_books: Dict[str, EterbaseOrderBook] = {}
+        self._saved_message_queues: Dict[str, Deque[EterbaseOrderBookMessage]] = defaultdict(lambda: deque(maxlen=1000))
+        self._active_order_trackers: Dict[str, EterbaseActiveOrderTracker] = defaultdict(EterbaseActiveOrderTracker)
+        self._trading_pairs: Optional[List[str]] = trading_pairs
+
+    @property
+    def data_source(self) -> OrderBookTrackerDataSource:
+        """
+        *required
+        Initializes an order book data source (Either from live API or from historical database)
+        :return: OrderBookTrackerDataSource
+        """
+        if not self._data_source:
+            if self._data_source_type is OrderBookTrackerDataSourceType.EXCHANGE_API:
+                self._data_source = EterbaseAPIOrderBookDataSource(trading_pairs=self._trading_pairs)
+            else:
+                raise ValueError(f"data_source_type {self._data_source_type} is not supported.")
+        return self._data_source
+
+    @property
+    def exchange_name(self) -> str:
+        """
+        *required
+        Name of the current exchange
+        """
+        return constants.EXCHANGE_NAME
+
+    async def _refresh_tracking_tasks(self):
+        """
+        Starts tracking for any new trading pairs, and stop tracking for any inactive trading pairs.
+        """
+        tracking_trading_pairs: Set[str] = set([key for key in self._tracking_tasks.keys()
+                                                if not self._tracking_tasks[key].done()])
+
+        available_pairs: Dict[str, EterbaseOrderBookTrackerEntry] = await self.data_source.get_tracking_pairs()
+        available_trading_pairs: Set[str] = set(available_pairs.keys())
+        new_trading_pairs: Set[str] = available_trading_pairs - tracking_trading_pairs
+        deleted_trading_pairs: Set[str] = tracking_trading_pairs - available_trading_pairs
+
+        for trading_pair in new_trading_pairs:
+            order_book_tracker_entry: EterbaseOrderBookTrackerEntry = available_pairs[trading_pair]
+            self._active_order_trackers[trading_pair] = order_book_tracker_entry.active_order_tracker
+            self._order_books[trading_pair] = order_book_tracker_entry.order_book
+            self._tracking_message_queues[trading_pair] = asyncio.Queue()
+            self._tracking_tasks[trading_pair] = safe_ensure_future(self._track_single_book(trading_pair))
+            self.logger().info("Started order book tracking for %s.", trading_pair)
```

Comment:
Minor typo in order book tracking log message.

<img width="455" alt="putty_RFPmlwxh2I" src="https://user-images.githubusercontent.com/50150287/84254631-0ea45d00-ab44-11ea-9bf6-48b7aec5b4e2.png">